config/file_config.go

```diff
@@ -311,11 +314,13 @@ func (f *fileConfig) validateSamplerConfigs() error {
 			}
 			err := f.rules.Unmarshal(i)
 			if err != nil {
+				logrus.WithError(err).Warn("Failed to unmarshal sampler rule")
```

Comment:
Looks like these validation errors get logged upstream. Would it make sense to add context to the returned error instead of adding a log statement, so that logging is consolidated to the topmost place?

Reply:
Yeah, good idea. Maybe something like this to wrap the error?

```golang
return fmt.Errorf("Failed to unmarshal sampler rule: %w", err)
```

I used this as [reference](https://blog.golang.org/go1.13-errors) on how to extend/wrap an `err`.

Reply:
That looks great!